### ISSUE TITLE
Age restriction on NFC ports

### ISSUE THREAD

**Issue by petterroea** [InfectedLan/InfectedAPI#220]
We should be able to disallow participants under a certain age from exiting during a given time. This should also be overrideable.

**Comment by halvors** [comment]
Are you proposing we add a `isImprisoned` boolean to the useroptions table?

**Comment by petterroea** [comment]
Yes :D

**Comment by halvors** [comment]
Hmm. So if a user is under 18 years old. Add a global option to the user. So not per event.

**Comment by petterroea** [comment]
15 and under

**Comment by petterroea** [comment]
specifically, they aren't allowed out between 1 and 7 am if their parents havent given written access to it

**Comment by halvors** [comment]
How should this option be set? By parents? How should we administrate that?

**Comment by halvors** [comment]
So under 15 is disallowed by default?

**Comment by petterroea** [comment]
People have to contact us, we set a flag on their account. We could do it like swimming. Yes, they are disallowed by default

**Comment by halvors** [comment]
So we just need a way to manually enable it for users whom we have a written concent from parens for?

**Comment by petterroea** [comment]
Correct.

**Comment by halvors** [comment]
You answered my question already.

**Comment by petterroea** [comment]
whoops

**Comment by petterroea** [comment]
I think the user api should just return if the user account is exit limited or not. The age limit should be in settings.php, and i'l handle the actual access control in nfc code

**Comment by halvors** [comment]
You want me to implement api stuff for this?

**Comment by halvors** [comment]
You want me to implement api stuff for this?

**Comment by petterroea** [comment]
i'l handle the nfc part, but i'd love if you could implement a user flag which specifies if a given user is allowed outside during lockdown hours or not(is set if parents contact us)

**Comment by halvors** [comment]
@petterroea Will you handle when lockdown hours is, if i provide an boolean for it on the user object?

**Comment by petterroea** [comment]
@halvors a boolean in user object that specifies if the user should be locked or not?

**Comment by halvors** [comment]
@petterroea Yeah, just based on age and wheater concent is present for user or not?

**Comment by petterroea** [comment]
yes, so the boolean should return age<age_limit(set in settings) && !allowedOutside, where allowedOutside is a flag settable by admins on accounts

**Comment by halvors** [comment]
@petterroea Create an entirely new table for "allowedOutside" flag, or use the "useroption" table for this?

**Comment by petterroea** [comment]
i think useroption

**Comment by halvors** [comment]
Yeah, think so as well.
